share/functions/__fish_edit_command_if_at_cursor.fish

```diff
@@ -0,0 +1,24 @@
+function __fish_edit_command_if_at_cursor --description 'If cursor is at the command token, edit the command source file'
+    set -l editor $argv
+    set -l tokens (commandline -xpc)
+    set -l command
+    if not set -q tokens[1]
+        set command (commandline -xp)[1]
+    else if test (count $tokens) = 1 && test -z "$(commandline -t)"
+        set command $tokens[1]
+    end
+
+    set -q command[1]
+    or return 1
+    set -l command_path (command -v -- $command)
+    or return 1
+    test -w $command_path
+    or return 1
+    string match -q 'text/*' (file --brief --mime-type -- $command_path)
```

Comment:
`file` would be a new dependency, we should probably error out if it doesn't exist.

An alternative is to look at the first line and see if it's a shebang.

Reply:
TIL `file` is POSIX but the specified version doesn't have any of the options we need.
But it seems everyone uses the same implementation (except for OpenBSD but it's still compatible).  
I'm leaning towards adding it to the dependencies, and print an error if it fails here. (We can still early-return though it makes no functional difference).

Reply:
In general long options aren't POSIX.

I'm *fine* with relying on file, I would just like a proper error here.